Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,301 @@
+
+            return answer;
+
+
+        }
+
+        private static decimal FindMorePeopleWithSameAmount(decimal HighestBalance, StringBuilder RichestPerson, string[] currentPersonData, decimal highestAmountOfPerson)
+        {
+            if (highestAmountOfPerson > HighestBalance)
+            {
+                HighestBalance = highestAmountOfPerson;
+                var currentOne = RichestPerson.ToString();
+
+                if (string.IsNullOrEmpty(currentOne))
+                {
+                    RichestPerson.Append(currentPersonData[0]);
+                }
+                else
+                {
+                    RichestPerson.Replace(currentOne, currentPersonData[0]);
+                }
+
+
+            }
+
+            return HighestBalance;
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var lowestBalance = int.MaxValue;
+            var poorestPerson = new StringBuilder();
+
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                if (!int.TryParse(currentPersonData[currentPersonData.Length - 1], out int highestAmountOfPerson))
+                {
+                    return "N/A.";
+                }
+
+                if (highestAmountOfPerson == lowestBalance)
+                {
+                    if (i == peopleAndBalances.Length - 1)
+                    {
+                        poorestPerson.Append(" and ");
+                    }
+                    else
+                    {
+                        poorestPerson.Append(", ");
+                    }
+
+                    poorestPerson.Append(currentPersonData[0]);
+                }
+
+                lowestBalance = FindMorePoorPeople(lowestBalance, poorestPerson, currentPersonData, highestAmountOfPerson);
+
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+            NumberFormatInfo noParens = (NumberFormatInfo)CultureInfo.CurrentCulture.NumberFormat.Clone();
+            noParens.CurrencyNegativePattern = 1; 
+            var lowestBalanceString = lowestBalance.ToString("C0", noParens);
+
+            var answer = $"{poorestPerson} has the least money. {lowestBalanceString}.";
+
+           return answer;
+
+
+        }
+
+        private static int FindMorePoorPeople(int lowestBalance, StringBuilder poorestPerson, string[] currentPersonData, int highestAmountOfPerson)
+        {
+            if (highestAmountOfPerson < lowestBalance)
+            {
+                lowestBalance = highestAmountOfPerson;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'lowestBalance'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4937470)